pkg/client/example_test.go

```diff
@@ -0,0 +1,179 @@
+	// Using a typed object.
+	p := &corev1.Pod{}
+	// c is a created client.
+	_ = c.Get(context.Background(), client.ObjectKey{
+		Namespace: "namespace",
+		Name:      "name",
+	}, p)
+
+	// Using a unstructured object.
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apps",
+		Kind:    "Deployment",
+		Version: "v1",
+	})
+	_ = c.Get(context.Background(), client.ObjectKey{
+		Namespace: "namespace",
+		Name:      "name",
+	}, u)
+}
+
+// This example shows how to use the client with typed and unstrucurted objects to create objects.
+func ExampleClient_create() {
+	// Using a typed object.
+	p := &corev1.Pod{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "namespace",
+			Name:      "name",
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				corev1.Container{
+					Image: "nginx",
+					Name:  "nginx",
+				},
+			},
+		},
+	}
+	// c is a created client.
+	_ = c.Create(context.Background(), p)
+
+	// Using a unstructured object.
+	u := &unstructured.Unstructured{}
+	u.Object = map[string]interface{}{
+		"name":      "name",
+		"namespace": "namespace",
+		"spec": map[string]interface{}{
+			"replicas": 2,
+			"selector": map[string]interface{}{
+				"matchLabels": map[string]interface{}{
+					"foo": "bar",
+				},
+			},
+			"template": map[string]interface{}{
+				"labels": map[string]interface{}{
+					"foo": "bar",
+				},
+				"spec": map[string]interface{}{
+					"containers": []map[string]interface{}{
+						{
+							"name":  "nginx",
+							"image": "nginx",
+						},
+					},
+				},
+			},
+		},
+	}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apps",
+		Kind:    "Deployment",
+		Version: "v1",
+	})
+	_ = c.Create(context.Background(), u)
+}
+
+// This example shows how to use the client with typed and unstrucurted objects to list objects.
+func ExampleClient_list() {
+	// Using a typed object.
+	p := &corev1.PodList{}
+	// c is a created client.
+	_ = c.List(context.Background(), nil, p)
+
+	// Using a unstructured object.
+	u := &unstructured.UnstructuredList{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apps",
+		Kind:    "DeploymentList",
+		Version: "v1",
+	})
+	_ = c.List(context.Background(), nil, u)
+}
+
+// This example shows how to use the client with typed and unstrucurted objects to update objects.
+func ExampleClient_update() {
+	// Using a typed object.
+	p := &corev1.Pod{}
+	// c is a created client.
+	_ = c.Get(context.Background(), client.ObjectKey{
+		Namespace: "namespace",
+		Name:      "name",
+	}, p)
+	p.SetFinalizers(append(p.GetFinalizers(), "new-finalizer"))
+	_ = c.Update(context.Background(), p)
+
+	// Using a unstructured object.
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apps",
+		Kind:    "Deployment",
+		Version: "v1",
+	})
+	_ = c.Get(context.Background(), client.ObjectKey{
+		Namespace: "namespace",
+		Name:      "name",
+	}, u)
```

Comment:
Can we rename `p` to `pod`, it helps with readability without increasing verbosity.